subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/CrossBuildInMemoryCacheFactory.java

```diff
@@ -43,6 +43,15 @@
      */
     <K, V> CrossBuildInMemoryCache<K, V> newCache();
 
+    /**
+     * Creates a new cache instance. Keys and values are always referenced using strong references.
+     *
+     * <p>Entries are only removed after each build session if they have not been used in this or the previous build.
+     *
+     * <p>Note: this should be used to create _only_ global/Gradle user home scoped instances.
+     */
+    <K, V> CrossBuildInMemoryCache<K, V> newCacheWithBuildAgedEviction();
```

Comment:
This name looks weird for me. What's a "build aged eviction"? I don't have a better name ready, but I think we should find one.